stdlib/http/src/main/java/org/ballerinalang/net/http/HttpConstants.java

```diff
@@ -395,8 +397,8 @@
     public static final String NATIVE_DATA_WEBSOCKET_CONNECTION_MANAGER = "NATIVE_DATA_WEBSOCKET_CONNECTION_MANAGER";
 
     public static final int REQUEST_STRUCT_INDEX = 1;
-    public static final String REQUEST_REUSE_STATUS_INDEX = "dirtyRequest";
     public static final boolean DIRTY_REQUEST = true;
+    public static final String NON_ENTITY_BODY_REQUEST = "NON_ENTITY_BODY_REQUEST";
```

Comment:
What I meant by NON_ENTITY_BODY_REQUEST is a request that does not have an entity body. Basically an entity consists of entity-header fields and an entity-body. Here the main requirement is to flag the requests which does not contain an entity-body to handle content-length/transfer-encoding headers.
If it is confusing, do you have any better suggestion to replace with?

Reply:
I would prefer a simpler term something like "REQUEST_WITHOUT_BODY" since the "Body" implicitly means the body of an entity, that is, the body of either a message or of a body part. 

To me, trouble with "NON_ENTITY_BODY_REQUEST" is when you read it aloud, it sounds weird. I guess it depends on how you read it, just like punctuations make all the difference to a meaning of a sentence. when I glanced it, I  automatically read it as "nonentity bodyrequest" which is totally weird but if you read it as "non entitybody request" it gives the proper meaning. 

Maybe I'm thinking too much :) , but I would personally prefer a simpler term that gives out the meaning at a glance.